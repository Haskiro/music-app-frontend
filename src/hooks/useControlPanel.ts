import { useState } from "react";
import { useAppDispatch } from "@store/hooks";

interface IUseControlPanelResult {
	searchValue: string;
	sortByAlphabet: boolean;
	onSearch: (searchValue: string) => void;
	onSort: () => void;
	onUpdate: () => void;
}

const useControlPanel = (fetchItems: any): IUseControlPanelResult => {
	const [searchValue, setSearchValue] = useState<string>("");
	const [sortByAlphabet, setSortByAlphabet] = useState<boolean>(false);
	const dispatch = useAppDispatch();

	const onSearch = (searchValue: string) => {
		setSearchValue(searchValue.toLowerCase());
	};
	const onSort = () => setSortByAlphabet((prevState) => !prevState);
	const onUpdate = () => dispatch(fetchItems());
	return { searchValue, sortByAlphabet, onSearch, onSort, onUpdate };
};

export default useControlPanel;
