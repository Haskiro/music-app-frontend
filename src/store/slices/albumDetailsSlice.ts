import { IAlbum } from "@interfaces/album.interface";
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import axios from "axios";

export interface albumDetailsState {
	album: IAlbum | null;
	status: "idle" | "loading" | "succeeded" | "failed";
}

const initialState: albumDetailsState = {
	album: null,
	status: "idle",
};

export const fetchAlbumById = createAsyncThunk(
	"albumDetails/fetchAlbumById",
	async (id: string) => {
		const response = await axios.get(
			`${process.env.REACT_APP_API}/albums/${id}`
		);
		return response.data;
	}
);

export const albumDetailsSlice = createSlice({
	name: "albumDetails",
	initialState,
	reducers: {},
	extraReducers(builder) {
		builder
			.addCase(fetchAlbumById.pending, (state) => {
				state.status = "loading";
			})
			.addCase(
				fetchAlbumById.fulfilled,
				(state, action: PayloadAction<IAlbum>) => {
					state.status = "succeeded";
					state.album = action.payload;
				}
			)
			.addCase(fetchAlbumById.rejected, (state) => {
				state.status = "failed";
			});
	},
});

// Action creators are generated for each case reducer function
export const {} = albumDetailsSlice.actions;

export default albumDetailsSlice.reducer;
