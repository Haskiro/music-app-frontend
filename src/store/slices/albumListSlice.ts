import { IAlbum } from "@interfaces/album.interface";
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import axios from "axios";

export interface albumListState {
	albumList: IAlbum[] | null;
	status: "idle" | "loading" | "succeeded" | "failed";
}

const initialState: albumListState = {
	albumList: null,
	status: "idle",
};

export const fetchAlbums = createAsyncThunk(
	"albumList/fetchAlbums",
	async () => {
		const response = await axios.get(`${process.env.REACT_APP_API}/albums`);
		return response.data;
	}
);

export const albumListSlice = createSlice({
	name: "artistList",
	initialState,
	reducers: {},
	extraReducers(builder) {
		builder
			.addCase(fetchAlbums.pending, (state) => {
				state.status = "loading";
			})
			.addCase(
				fetchAlbums.fulfilled,
				(state, action: PayloadAction<IAlbum[]>) => {
					state.status = "succeeded";
					state.albumList = action.payload;
				}
			)
			.addCase(fetchAlbums.rejected, (state) => {
				state.status = "failed";
			});
	},
});

// Action creators are generated for each case reducer function
export const {} = albumListSlice.actions;

export default albumListSlice.reducer;
