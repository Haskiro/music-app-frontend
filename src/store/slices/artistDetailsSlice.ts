import { IArtist } from "@interfaces/artist.interface";
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import axios from "axios";

export interface artistDetailsState {
	artist: IArtist | null;
	status: "idle" | "loading" | "succeeded" | "failed";
}

const initialState: artistDetailsState = {
	artist: null,
	status: "idle",
};

export const fetchArtistById = createAsyncThunk(
	"artistDetails/fetchArtist",
	async (id: string) => {
		const response = await axios.get(
			`${process.env.REACT_APP_API}/artists/${id}`
		);
		return response.data;
	}
);

export const artistDetailsSlice = createSlice({
	name: "artistDetails",
	initialState,
	reducers: {},
	extraReducers(builder) {
		builder
			.addCase(fetchArtistById.pending, (state) => {
				state.status = "loading";
			})
			.addCase(
				fetchArtistById.fulfilled,
				(state, action: PayloadAction<IArtist>) => {
					state.status = "succeeded";
					state.artist = action.payload;
				}
			)
			.addCase(fetchArtistById.rejected, (state) => {
				state.status = "failed";
			});
	},
});

// Action creators are generated for each case reducer function
export const {} = artistDetailsSlice.actions;

export default artistDetailsSlice.reducer;
