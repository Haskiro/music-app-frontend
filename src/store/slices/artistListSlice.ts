import { IArtist } from "@interfaces/artist.interface";
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import axios from "axios";

export interface artistListState {
	artistList: IArtist[];
	status: "idle" | "loading" | "succeeded" | "failed";
}

const initialState: artistListState = {
	artistList: [],
	status: "idle",
};

export const fetchArtists = createAsyncThunk(
	"artistList/fetchArtists",
	async () => {
		const response = await axios.get(
			`${process.env.REACT_APP_API}/artists`
		);
		return response.data;
	}
);

export const artistListSlice = createSlice({
	name: "artistList",
	initialState,
	reducers: {},
	extraReducers(builder) {
		builder
			.addCase(fetchArtists.pending, (state) => {
				state.status = "loading";
			})
			.addCase(
				fetchArtists.fulfilled,
				(state, action: PayloadAction<IArtist[]>) => {
					state.status = "succeeded";
					state.artistList = action.payload;
				}
			)
			.addCase(fetchArtists.rejected, (state) => {
				state.status = "failed";
			});
	},
});

// Action creators are generated for each case reducer function
export const {} = artistListSlice.actions;

export default artistListSlice.reducer;
