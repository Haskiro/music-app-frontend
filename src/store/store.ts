import { configureStore, combineReducers } from "@reduxjs/toolkit";
import {
	persistStore,
	persistReducer,
	FLUSH,
	REHYDRATE,
	PAUSE,
	PERSIST,
	PURGE,
	REGISTER,
} from "redux-persist";
import storage from "redux-persist/lib/storage";
import userReducer from "./slices/userSlice";

const persistConfig = {
	key: "user",
	version: 1,
	storage,
	whitelist: ["user"],
};

const userPersistConfig = {
	key: "accessToken",
	storage,
	whitelist: ["accessToken"],
};

const rootReducer = combineReducers({
	user: persistReducer(userPersistConfig, userReducer),
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
	reducer: persistedReducer,
	middleware: (getDefaultMiddleware) =>
		getDefaultMiddleware({
			serializableCheck: {
				ignoredActions: [
					FLUSH,
					REHYDRATE,
					PAUSE,
					PERSIST,
					PURGE,
					REGISTER,
				],
			},
		}),
});

export let persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
