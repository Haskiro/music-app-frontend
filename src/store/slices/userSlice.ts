import { IAccessToken, IUser } from "@interfaces/user.interface";
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import { AppDispatch, RootState } from "@store/store";
import axios from "axios";

export interface UserState {
	user: IUser | null;
	accessToken: string | null;
	status: "idle" | "loading" | "succeeded" | "failed";
	authStatus: "idle" | "loading" | "succeeded" | "failed";
}

export interface ILoginData {
	email: string;
	password: string;
}

export interface IRegisterData extends ILoginData {
	last_name: string;
	first_name: string;
}

const initialState: UserState = {
	user: null,
	accessToken: null,
	status: "idle",
	authStatus: "idle",
};

export const authUser = createAsyncThunk(
	"user/authUser",
	async (loginData: ILoginData) => {
		const response = await axios.post(
			`${process.env.REACT_APP_API}/auth/login/`,
			loginData,
			{
				headers: {
					"Content-Type": "application/json",
				},
			}
		);
		return response.data;
	}
);

export const registerUser = createAsyncThunk(
	"user/registerUser",
	async (registerData: IRegisterData) => {
		const response = await axios.post(
			`${process.env.REACT_APP_API}/auth/register/`,
			registerData,
			{
				headers: {
					"Content-Type": "application/json",
				},
			}
		);
		return response.data;
	}
);

export const fetchUser = createAsyncThunk<
	IUser,
	undefined,
	{ state: RootState }
>("user/fetchUser", async (_, thunkApi) => {
	const state = thunkApi.getState();
	const response = await axios.get(`${process.env.REACT_APP_API}/auth/me/`, {
		headers: {
			Authorization: "Bearer " + state.user.accessToken,
		},
	});
	return response.data.photo
		? {
				...response.data,
				photo: `${process.env.REACT_APP_DOMAIN}${response.data.photo}`,
		  }
		: response.data;
});

export const updateUser = createAsyncThunk<
	{ message: string },
	IUser,
	{ dispatch: AppDispatch; state: RootState }
>("user/updateUser", async (updateData, thunkApi) => {
	const state = thunkApi.getState();
	const response = await axios.patch(
		`${process.env.REACT_APP_API}/auth/update/`,
		updateData,
		{
			headers: {
				"Content-Type": "application/json",
				Authorization: "Bearer " + state.user.accessToken,
			},
		}
	);
	const dispatch = thunkApi.dispatch;
	dispatch(editUser(updateData));

	return response.data;
});

export const userSlice = createSlice({
	name: "user",
	initialState,
	reducers: {
		logout: (state) => {
			state.accessToken = null;
			state.user = null;
			state.status = "idle";
			state.authStatus = "idle";
		},
		editUser: (state, action: PayloadAction<IUser>) => {
			state.user = action.payload;
		},
	},
	extraReducers(builder) {
		builder
			.addCase(authUser.pending, (state) => {
				state.authStatus = "loading";
			})
			.addCase(
				authUser.fulfilled,
				(state, action: PayloadAction<IAccessToken>) => {
					state.authStatus = "succeeded";
					state.accessToken = action.payload.access;
				}
			)
			.addCase(authUser.rejected, (state) => {
				state.authStatus = "failed";
			})
			.addCase(registerUser.pending, (state) => {
				state.authStatus = "loading";
			})
			.addCase(
				registerUser.fulfilled,
				(state, action: PayloadAction<IAccessToken>) => {
					state.authStatus = "succeeded";
					state.accessToken = action.payload.access;
				}
			)
			.addCase(registerUser.rejected, (state) => {
				state.authStatus = "failed";
			})
			.addCase(fetchUser.pending, (state) => {
				state.status = "loading";
			})
			.addCase(
				fetchUser.fulfilled,
				(state, action: PayloadAction<IUser>) => {
					state.status = "succeeded";
					state.user = action.payload;
				}
			)
			.addCase(fetchUser.rejected, (state) => {
				state.status = "failed";
			})
			.addCase(updateUser.pending, (state) => {
				state.status = "loading";
			})
			.addCase(updateUser.fulfilled, (state) => {
				state.status = "succeeded";
			})
			.addCase(updateUser.rejected, (state) => {
				state.status = "failed";
			});
	},
});

// Action creators are generated for each case reducer function
export const { logout, editUser } = userSlice.actions;

export default userSlice.reducer;
