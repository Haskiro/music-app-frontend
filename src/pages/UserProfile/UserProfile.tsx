import { FC, useEffect } from "react";
import UserPhotoPlug from "@assets/images/user-photo-plug.png";
import { IUser } from "@interfaces/user.interface";
import { useAppDispatch, useAppSelector } from "@store/hooks";
import { fetchUser, updateUser } from "@store/slices/userSlice";
import { Button, Form, Input, Spin } from "antd";
import styles from "./UserProfile.module.scss";

const UserProfile: FC = () => {
	const { TextArea } = Input;
	const dispatch = useAppDispatch();
	const user = useAppSelector((state) => state.user.user);
	const status = useAppSelector((state) => state.user.status);

	const [form] = Form.useForm();

	const onFill = () => {
		form.setFieldsValue({
			email: user?.email,
			name: user?.first_name,
			surname: user?.last_name,
			bio: user?.bio,
		});
	};

	const onFinish = (fields: {
		email: string;
		name: string;
		surname: string;
		bio: string;
	}) => {
		dispatch(
			updateUser({
				email: fields.email,
				first_name: fields.name,
				last_name: fields.surname,
				bio: fields.bio,
				photo: user?.photo || "",
			})
		);
	};

	useEffect(() => {
		dispatch(fetchUser());
		onFill();
	}, []);

	const content = (
		<>
			<div className={styles.header}>
				<img
					src={user?.photo ? user.photo : UserPhotoPlug}
					alt="Фотография"
					className={styles.photo}
					height="250"
					width="250"
				/>
				<div className={styles.block}>
					<h1 className={styles.title}>Профиль</h1>
					<p className={styles.text}>
						{user?.first_name} {user?.last_name}
					</p>
				</div>
			</div>
			<div className={styles.body}>
				<p className={styles.text}>
					{user?.bio ? user.bio : "Нет заполненной биографии"}
				</p>
			</div>
			<h2 className={styles.formHeading}>Редактировать Профиль</h2>
			<Form
				name="control-ref"
				labelCol={{ span: 3 }}
				wrapperCol={{ span: 16 }}
				style={{ marginTop: "20px" }}
				form={form}
				onFinish={onFinish}
			>
				<Form.Item name="email" label="Почта">
					<Input value={user?.first_name} />
				</Form.Item>
				<Form.Item name="name" label="Имя">
					<Input value={user?.first_name} />
				</Form.Item>
				<Form.Item name="surname" label="Фамилия">
					<Input value={user?.last_name} />
				</Form.Item>
				<Form.Item name="bio" label="Биография">
					<TextArea rows={4} value={user?.bio} />
				</Form.Item>
				<Form.Item wrapperCol={{ offset: 3, span: 16 }}>
					<Button type="primary" htmlType="submit">
						Отправить
					</Button>
					<Button type="link" htmlType="button" onClick={onFill}>
						Отменить изменения
					</Button>
				</Form.Item>
			</Form>
		</>
	);

	return (
		<div className={styles.container}>
			{status === "loading" ? (
				<Spin
					tip="Loading"
					size="large"
					style={{
						margin: "0px auto",
						display: "block",
						marginTop: "20px",
					}}
				/>
			) : null}
			{status === "succeeded" ? content : null}
		</div>
	);
};

export default UserProfile;
