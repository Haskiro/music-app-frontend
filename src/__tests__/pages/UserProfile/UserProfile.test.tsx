import { IUser } from "@interfaces/user.interface";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import UserProfile from "../../../pages/UserProfile";
import * as reduxHooks from "../../../store/hooks";
import * as actions from "../../../store/slices/userSlice";

jest.mock("../../../store/hooks");
jest.mock("antd", () => {
	const originalModule = jest.requireActual("antd");
	return {
		__esModule: true,
		...originalModule,
		Spin: () => <div data-testid="Spin" />,
	};
});
Object.defineProperty(window, "matchMedia", {
	writable: true,
	value: jest.fn().mockImplementation((query) => ({
		matches: false,
		media: query,
		onchange: null,
		addListener: jest.fn(), // Deprecated
		removeListener: jest.fn(), // Deprecated
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
		dispatchEvent: jest.fn(),
	})),
});

const mockedUseAppDispatch = jest.spyOn(reduxHooks, "useAppDispatch");
const mockedUseAppSelector = jest.spyOn(reduxHooks, "useAppSelector");
const mockedFetchUser = jest.spyOn(actions, "fetchUser");
const mockedUpdateUser = jest.spyOn(actions, "updateUser");

describe("User Profile", () => {
	const mockUser: IUser = {
		first_name: "Pavel",
		last_name: "Surname",
		bio: "",
		email: "email@email.email",
		photo: "",
	};

	beforeEach(() => {
		jest.clearAllMocks();
	});

	it("should create user profile page with user data", () => {
		mockedUseAppDispatch.mockReturnValue(jest.fn());
		mockedUseAppSelector.mockReturnValueOnce(mockUser);
		mockedUseAppSelector.mockReturnValueOnce("succeeded");

		const page = render(<UserProfile />);

		expect(page).toMatchSnapshot();
	});

	it("should render spinner", () => {
		mockedUseAppDispatch.mockReturnValue(jest.fn());
		mockedUseAppSelector.mockReturnValueOnce({});
		mockedUseAppSelector.mockReturnValueOnce("loading");

		render(<UserProfile />);

		expect(screen.getByTestId("Spin")).toBeInTheDocument();
	});

	it("should dispatch action fetchUser", () => {
		const dispatch = jest.fn();
		mockedUseAppDispatch.mockReturnValue(dispatch);
		mockedUseAppSelector.mockReturnValueOnce(mockUser);
		mockedUseAppSelector.mockReturnValueOnce("succeeded");

		render(<UserProfile />);

		expect(dispatch).toHaveBeenCalledTimes(1);
		expect(mockedFetchUser).toHaveBeenCalledTimes(1);
	});

	it("should dispatch action updateUser on clicking submit button", async () => {
		const user = userEvent.setup();
		const dispatch = jest.fn();
		mockedUseAppDispatch.mockReturnValue(dispatch);
		mockedUseAppSelector.mockReturnValueOnce(mockUser);
		mockedUseAppSelector.mockReturnValueOnce("succeeded");

		render(<UserProfile />);
		const button = screen.getByText("Отправить");
		await user.click(button);

		expect(dispatch).toHaveBeenCalledTimes(2);
		expect(mockedUpdateUser).toHaveBeenCalledTimes(1);
		expect(mockedUpdateUser).toHaveBeenCalledWith({
			...mockUser,
			photo: "",
		});
	});

	it("should set form field with default values on clicking reset button", async () => {
		const user = userEvent.setup();
		const dispatch = jest.fn();
		mockedUseAppDispatch.mockReturnValue(dispatch);
		mockedUseAppSelector.mockReturnValueOnce(mockUser);
		mockedUseAppSelector.mockReturnValueOnce("succeeded");

		render(<UserProfile />);
		const button = screen.getByText("Отменить изменения");
		const nameField = screen.getByLabelText("Имя");
		const surnameField = screen.getByLabelText("Фамилия");

		await user.type(nameField, "Ivan");
		await user.type(surnameField, "Ivanov");
		await user.click(button);

		expect(nameField).toHaveValue(mockUser.first_name);
		expect(surnameField).toHaveValue(mockUser.last_name);
	});
});
