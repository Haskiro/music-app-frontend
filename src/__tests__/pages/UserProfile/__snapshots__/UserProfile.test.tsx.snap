// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Profile should create user profile page with user data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header"
        >
          <img
            alt="Фотография"
            class="photo"
            height="250"
            src="user-photo-plug.png"
            width="250"
          />
          <div
            class="block"
          >
            <h1
              class="title"
            >
              Профиль
            </h1>
            <p
              class="text"
            >
              Pavel
               
              Surname
            </p>
          </div>
        </div>
        <div
          class="body"
        >
          <p
            class="text"
          >
            Нет заполненной биографии
          </p>
        </div>
        <h2
          class="formHeading"
        >
          Редактировать Профиль
        </h2>
        <form
          class="ant-form ant-form-horizontal css-dev-only-do-not-override-sk7ap8"
          id="control-ref"
          style="margin-top: 20px;"
        >
          <div
            class="ant-form-item css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-col ant-col-3 ant-form-item-label css-dev-only-do-not-override-sk7ap8"
              >
                <label
                  class=""
                  for="control-ref_email"
                  title="Почта"
                >
                  Почта
                </label>
              </div>
              <div
                class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input css-dev-only-do-not-override-sk7ap8"
                      id="control-ref_email"
                      type="text"
                      value="email@email.email"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-col ant-col-3 ant-form-item-label css-dev-only-do-not-override-sk7ap8"
              >
                <label
                  class=""
                  for="control-ref_name"
                  title="Имя"
                >
                  Имя
                </label>
              </div>
              <div
                class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input css-dev-only-do-not-override-sk7ap8"
                      id="control-ref_name"
                      type="text"
                      value="Pavel"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-col ant-col-3 ant-form-item-label css-dev-only-do-not-override-sk7ap8"
              >
                <label
                  class=""
                  for="control-ref_surname"
                  title="Фамилия"
                >
                  Фамилия
                </label>
              </div>
              <div
                class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input css-dev-only-do-not-override-sk7ap8"
                      id="control-ref_surname"
                      type="text"
                      value="Surname"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-col ant-col-3 ant-form-item-label css-dev-only-do-not-override-sk7ap8"
              >
                <label
                  class=""
                  for="control-ref_bio"
                  title="Биография"
                >
                  Биография
                </label>
              </div>
              <div
                class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <textarea
                      class="ant-input css-dev-only-do-not-override-sk7ap8"
                      id="control-ref_bio"
                      rows="4"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-col ant-col-16 ant-col-offset-3 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-sk7ap8 ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        Отправить
                      </span>
                    </button>
                    <button
                      class="ant-btn css-dev-only-do-not-override-sk7ap8 ant-btn-link"
                      type="button"
                    >
                      <span>
                        Отменить изменения
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <img
          alt="Фотография"
          class="photo"
          height="250"
          src="user-photo-plug.png"
          width="250"
        />
        <div
          class="block"
        >
          <h1
            class="title"
          >
            Профиль
          </h1>
          <p
            class="text"
          >
            Pavel
             
            Surname
          </p>
        </div>
      </div>
      <div
        class="body"
      >
        <p
          class="text"
        >
          Нет заполненной биографии
        </p>
      </div>
      <h2
        class="formHeading"
      >
        Редактировать Профиль
      </h2>
      <form
        class="ant-form ant-form-horizontal css-dev-only-do-not-override-sk7ap8"
        id="control-ref"
        style="margin-top: 20px;"
      >
        <div
          class="ant-form-item css-dev-only-do-not-override-sk7ap8"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-col ant-col-3 ant-form-item-label css-dev-only-do-not-override-sk7ap8"
            >
              <label
                class=""
                for="control-ref_email"
                title="Почта"
              >
                Почта
              </label>
            </div>
            <div
              class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input css-dev-only-do-not-override-sk7ap8"
                    id="control-ref_email"
                    type="text"
                    value="email@email.email"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item css-dev-only-do-not-override-sk7ap8"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-col ant-col-3 ant-form-item-label css-dev-only-do-not-override-sk7ap8"
            >
              <label
                class=""
                for="control-ref_name"
                title="Имя"
              >
                Имя
              </label>
            </div>
            <div
              class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input css-dev-only-do-not-override-sk7ap8"
                    id="control-ref_name"
                    type="text"
                    value="Pavel"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item css-dev-only-do-not-override-sk7ap8"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-col ant-col-3 ant-form-item-label css-dev-only-do-not-override-sk7ap8"
            >
              <label
                class=""
                for="control-ref_surname"
                title="Фамилия"
              >
                Фамилия
              </label>
            </div>
            <div
              class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input css-dev-only-do-not-override-sk7ap8"
                    id="control-ref_surname"
                    type="text"
                    value="Surname"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item css-dev-only-do-not-override-sk7ap8"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-col ant-col-3 ant-form-item-label css-dev-only-do-not-override-sk7ap8"
            >
              <label
                class=""
                for="control-ref_bio"
                title="Биография"
              >
                Биография
              </label>
            </div>
            <div
              class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <textarea
                    class="ant-input css-dev-only-do-not-override-sk7ap8"
                    id="control-ref_bio"
                    rows="4"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item css-dev-only-do-not-override-sk7ap8"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-sk7ap8"
          >
            <div
              class="ant-col ant-col-16 ant-col-offset-3 ant-form-item-control css-dev-only-do-not-override-sk7ap8"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-sk7ap8 ant-btn-primary"
                    type="submit"
                  >
                    <span>
                      Отправить
                    </span>
                  </button>
                  <button
                    class="ant-btn css-dev-only-do-not-override-sk7ap8 ant-btn-link"
                    type="button"
                  >
                    <span>
                      Отменить изменения
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
